### DISCUSSION TITLE
Is it possible to run QEMU raw images or QEMU qcow2 images in firecracker?

### DISCUSSION THREAD

**Discussion by lolwierd** [firecracker-microvm/firecracker#3873] [Q&A] [answered]
I tried extracting the rootfs from a qemu image and mounting it in firecracker but it didnt work.
I also tried just mounting the raw image as rootfs but that didn't work as well. Maybe I just need to recompile my custom kernal with some specific params? Or is it just not possible?

Any help is highly appreciated! Thank you!

**Comment by gbionescu** [comment]
Generally it should work, but the how-to on doing this is not straight forward as in depends on how the original rootfs is designed to be booted up, which is where things start getting more complex - you can have initrd images or self-unpacking filesystems.

Short answer: it's possible, but there are a lot of variables.

Try looking into the kernel startup sequence as it usually gives you specific errors on what's not working properly.

**Comment by kalyazin** [comment, marked as answer]
@LOLwierd thanks for your enquiry. No, that is not supported in Firecracker at the moment and we have not investigated what it would entail.
